### PR TITLE
use AppM

### PR DESCRIPTION
**Pull request by shmish111**
Create a special monad for the dashboard and use it to make our first call to the PAB HTTP API

Pre-submit checklist:
- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: marlowe-dashboard-client/src/Main.purs [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
 import Control.Coroutine (Consumer, Process, connect, consumer, runProcess)
 import Data.Maybe (Maybe(..))
 import Effect (Effect)
 import Effect.Aff (Aff, forkAff)
 import Effect.Class (liftEffect)
 import Effect.Console (log)
 import Effect.Unsafe (unsafePerformEffect)
+import Env (Env)
 import Foreign.Generic (defaultOptions)
+import Halogen as H
```
Comment (merivale) [inline marlowe-dashboard-client/src/Main.purs:14]: small lint thing, but why not just import `hoist` and `Component` explicitly?

File: marlowe-dashboard-client/src/Main.purs [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
 import Halogen.Aff (awaitBody, runHalogenAff)
+import Halogen.HTML as HH
```
Comment (merivale) [inline marlowe-dashboard-client/src/Main.purs:16]: likewise could just import `HTML`

**Review by merivale — APPROVED** [review]
LGTM

File: marlowe-dashboard-client/src/Contract/State.purs [hunk 3]
```diff
@@ -30,14 +40,24 @@ handleQuery (ApplyTx tx next) = do
...
+  mContractId <- use _contractId
+  for_ mContractId \contractId -> do
+    let
+      txInput = mkTx currentExeState (Unfoldable.fromMaybe input)
 
-    executionState = nextState currentExeState txInput
-  assign _executionState executionState
+      json = RawJson <<< unsafeStringify <<< encode $ input
+    res <- mapEnvReaderT _.ajaxSettings $ runExceptT $ postApiContractByContractinstanceidEndpointByEndpointname json contractId "apply-inputs"
+    for_ res \_ -> do
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Contract/State.purs:57]: runMaybeT?

**Review by hrajchert — APPROVED** [review]
LGTM

**Pull request merged**